reactive/etcd.py

```diff
@@ -154,10 +155,24 @@ def remove_states():
     remove_state('etcd.ssl.exported')
 
 
+@hook('pre-series-upgrade')
+def pre_series_upgrade():
+    bag = EtcdDatabag()
+    host.service_pause(bag.etcd_daemon)
+    status_set('blocked', 'Series upgrade in progress')
```

Comment:
Seems like this status could easily get overwritten by other handlers. I recommend setting it in an atexit handler or refactoring the charm to use layer-status.